app/(routes)/CreateScales/page.tsx

```diff
@@ -0,0 +1,186 @@
+"use client";
+import { modifyNotesActionTypes } from "../../lib/actionTypes";
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useReducer,
+  useRef,
+  useState,
+} from "react";
+import VexFlow from "vexflow";
+import BlueButton from "../../components/BlueButton";
+import CheckIfNoteFound from "../../components/CheckIfNoteFound";
+import CheckNumBeatsInMeasure from "../../components/CheckNumBeatsInMeasure";
+import { buttonGroup } from "../../lib/buttonsAndButtonGroups";
+import { initialNotesAndCoordsState } from "../../lib/data/initialNotesAndCoordinatesState";
+import { staveData } from "../../lib/data/stavesData";
+import { findBarIndex } from "../../lib/findBar";
+import generateYMinAndYMaxForAllNotes from "../../lib/generateYMinAndMaxForAllNotes";
+import getUserClickInfo from "../../lib/getUserClickInfo";
+import { HandleScaleInteraction } from "../../lib/handleScaleInteraction";
+import { noteInteractionInitialState } from "../../lib/initialStates";
+import { initializeRenderer } from "../../lib/initializeRenderer";
+import { notesArray } from "../../lib/noteArray";
+import { scaleReducer } from "../../lib/reducer";
+import { setupRendererAndDrawNotes } from "../../lib/setupRendererAndDrawNotesNewest";
+import { ScaleData, StaveType } from "../../lib/typesAndInterfaces";
+
+const { Renderer } = VexFlow.Flow;
+
+const ManageScales = () => {
+  const rendererRef = useRef<InstanceType<typeof Renderer> | null>(null);
+  const container = useRef<HTMLDivElement | null>(null);
+  const [staves, setStaves] = useState<StaveType[]>([]);
+  const [scaleDataMatrix, setScaleDataMatrix] = useState<ScaleData[][]>([[]]);
+  const [notesAndCoordinates, setNotesAndCoordinates] = useState([
+    initialNotesAndCoordsState,
+  ]);
+
+  const [state, dispatch] = useReducer(
+    scaleReducer,
+    noteInteractionInitialState
+  );
+
+  const noNoteFound = () => dispatch({ type: "noNoteFound" });
+
+  const tooManyBeatsInMeasure = () =>
+    dispatch({ type: "tooManyBeatsInMeasure" });
+
+  const modifyStaveNotesButtonGroup = useMemo(
+    () => buttonGroup(dispatch, state, modifyNotesActionTypes),
+    [dispatch, state]
+  );
+
+  const eraseMeasures = () => {
+    setScaleDataMatrix((): ScaleData[][] => {
+      return [[]];
+    });
+    setNotesAndCoordinates(() =>
+      generateYMinAndYMaxForAllNotes(147, notesArray)
+    );
+    renderStavesAndNotes();
+  };
+
+  const renderStavesAndNotes = useCallback(
+    (): void =>
+      setupRendererAndDrawNotes({
+        rendererRef,
+        ...staveData,
+        setStaves,
+        scaleDataMatrix,
+        staves,
+      }),
+    [rendererRef, setStaves, scaleDataMatrix, staves]
+  );
+
+  useEffect(() => {
```

Comment:
Maybe lines 77-86 can be combined to one useEffect since both are triggered by empty dependency arrays